verilog/analysis/symbol_table.cc

```diff
@@ -582,6 +608,29 @@ class SymbolTable::Builder : public TreeContextVisitor {
       return;
     }
 
+    // Handle possible implicit declarations here
+    if (declaration_type_info_ != nullptr && declaration_type_info_->implicit) {
+      const SymbolTableNode* resolved =
+          LookupSymbolUpwards(*ABSL_DIE_IF_NULL(current_scope_), text);
+      if (resolved == nullptr) {
+        // No explicit declaration found, declare here
+        SymbolTableNode& implicit_declaration =
+            EmplaceTypedElementInCurrentScope(
+                leaf, text, SymbolMetaType::kDataNetVariableInstance);
+
+        const ReferenceComponent implicit_ref{
+            .identifier = text,
+            .ref_type = InferReferenceType(),
+            .required_metatype = InferMetaType(),
+            // pre-resolve
+            .resolved_symbol = &implicit_declaration,
+        };
+
+        ref.PushReferenceComponent(implicit_ref);
```

Comment:
Is this push needed?  When a symbol is implicitly declared, do we ever need to build a hierarchical reference from it with the net/variable as a parent?

Reply:
I think we do. Implicit declaration is also an reference to that symbol and we want to keep that reference.